plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,485 @@
+  report_link_local_groups:
+    description:
+    - Enable or disable reporting link-local groups.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  igmp_version:
+    description:
+    - The IGMP version of the IGMP Interface Policy.
+    - Defaults to C(v2) when unset during creation.
+    type: str
+    choices: [ v2, v3 ]
+  group_timeout:
+    description:
+    - The group timeout value in seconds.
+    - Defaults to 260 when unset during creation.
+    - The valid range is from 3 to 65535.
+    type: int
+  query_interval:
+    description:
+    - The query interval value in seconds.
+    - Defaults to 125 when unset during creation.
+    - The valid range is from 1 to 18000.
+    type: int
+  query_response_interval:
+    description:
+    - The query response interval value in seconds.
+    - Defaults to 10 when unset during creation.
+    - The valid range is from 1 to 25.
+    type: int
+  last_member_count:
+    description:
+    - The last member query count value.
+    - Defaults to 2 when unset during creation.
+    - The valid range is from 1 to 5.
+    type: int
+  last_member_response_time:
+    description:
+    - The last member query response time value in seconds.
+    - Defaults to 1 when unset during creation.
+    - The valid range is from 1 to 25.
+    type: int
+  startup_query_count:
+    description:
+    - The startup query count value.
+    - Defaults to 2 when unset during creation.
+    - The valid range is from 1 to 10.
+    type: int
+  startup_query_interval:
+    description:
+    - The startup query interval value in seconds.
+    - Defaults to 31 when unset during creation.
+    - The valid range is from 1 to 18000.
+    type: int
+  querier_timeout:
+    description:
+    - The querier timeout value in seconds.
+    - Defaults to 255 when unset during creation.
+    - The valid range is from 1 to 65535.
+    type: int
+  robustness_variable:
+    description:
+    - The robustness variable value.
+    - Defaults to 2 when unset during creation.
+    - The valid range is from 1 to 7.
+    type: int
+  state_limit_route_map_uuid:
+    description:
+    - The UUID of the state limit route map.
+    type: str
+    aliases: [ state_limit_route_map_for_multicast ]
+  state_limit_route_map:
+    description:
+    - The Route Map Policy for Multicast.
+    - This parameter can be used instead of O(state_limit_route_map_uuid).
+    - If both parameter are used, O(state_limit_route_map) will be ignored.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the Route Map Policy for Multicast.
+        type: str
+        required: true
+    aliases: [ state_limit_route_map_policy, state_limit_route_map_policy_multicast ]
+  report_policy_route_map_uuid:
+    description:
+    - The UUID of the report policy route map.
+    type: str
+    aliases: [ report_policy_route_map_for_multicast ]
+  report_policy_route_map:
```

Comment:
How do we delete route-maps from the configuration?
Why is report_policy_route_map a dict with only 1 parameter, is string not easier in this case for name?

Reply:
at this point the object only support report_policy_route_map from the same template, but in future there is a chance that it might use report_policy_route_map from a different template. Hence after the discussion, it was decided to retain it as a map

Reply:
Made changes to support, removal of policy ref when dict is {} or uuid = ""